lib/xandra/protocol.ex

```diff
@@ -1,6 +1,26 @@
 defmodule Xandra.Protocol do
   @moduledoc false
 
+  defmacrop decode_string(buffer, value) do
+    quote do
+      <<size::16, unquote(value)::size(size)-bytes, unquote(buffer)::bits>> = unquote(buffer)
+    end
+  end
+
+  defmacrop decode_value(buffer, value, type, do: block) do
```

Comment:
Out of curiosity, could you measure the impact of replacing the "do" block here with a function?

```elixir
decode_value(buffer, value, type, fn value -> ... end)
```

? There is *a lot* of magic going on here with the magic `value` variable being assigned before the block and so on. If the impact is negligible, I'd personally go with `fn`.